src/wasm/wasm-binary.cpp

```diff
@@ -795,6 +795,33 @@ void WasmBinaryWriter::writeNames() {
   // TODO: label, type, and element names
   // see: https://github.com/WebAssembly/extended-name-section
 
+  // GC field names
+  if (wasm->features.hasGC()) {
+    std::vector<HeapType> namedTypes;
+    for (auto& type : types) {
+      if (type.isStruct() && wasm->typeNames.count(type) &&
+          !wasm->typeNames.at(type).fieldNames.empty()) {
```

Comment:
```suggestion
          !wasm->typeNames[type].fieldNames.empty()) {
```
I don't think there's any reason to use `.at()` instead of `[]` here, since we know this will never fail. Same below.

Reply:
I think `.at` is useful here because it cannot modify the underlying data structure. That is, `X[Y]` will potentially create an element.

While not strictly needed here, I think it improves readability. It would also allow the code to compile if the data structure were made `const` at some point.

Reply:
We know that `[]` will not perform any modifications in this case because we've already verified that `type` is present in the collection with `count`. Also, `[]` still works on const collections; it just returns a const reference instead of a mutable reference.

Anyway, I don't feel strongly about this and I agree that `.at` is generally better than `[]`. I was just surprised because generally we use `[]` over `.at` in similar situations. It would be nice if we could be more consistent in our usage. Maybe clang-tidy can help?

Reply:
I agree we should be more consistent. I can't remember offhand where this pattern happens, and it's not easy to grep for, though, so yeah, if tidy can help that could be great.

> Also, [] still works on const collections;

The problem is that if the map itself is const, we error on `[]` as it may insert a new element. Concrete example:
```cpp
#include <map>

int main() {
  const std::map<int, int> set;
  set[10];
}
```
```
$ g++ a.cpp
a.cpp: In function ‘int main()’:
a.cpp:5:9: error: passing ‘const std::map<int, int>’ as ‘this’ argument discards qualifiers [-fpermissive]
   set[10];
```

Reply:
Oh yeah, I was wrong about `[]` working when `const`. That's very sad.